### ISSUE TITLE
Accuracy on training set of Kinetics dataset using RGB pre-trained model

### ISSUE THREAD

**Issue by xhappy** [qijiezhao/pseudo-3d-pytorch#5]
Hello Qijie,

Now, I really need you help now...since I just tried your pre-trained model on Kinetics dataset is low....

My codes are as following:

parser = argparse.ArgumentParser(description='video(< 17 second) label prediction')
    
    parser.add_argument('video_filename')
    
    # start_time = time.time()
    
    args = parser.parse_args()
    
    yt_vid, extension = args.video_filename.split('/')[-1].split('.')
    
    vid_reader = imageio.get_reader(args.video_filename, 'ffmpeg')
    img_list = get_img_list_from_vid_reader(vid_reader, extension)
    img_list = img_list.transpose(3, 0, 1, 2)
    
    img_list = np.expand_dims(img_list, axis = 0)

    model = P3D199(pretrained=True,num_classes=400)
    model = model.cuda()
    # data=torch.autograd.Variable(torch.rand(10,3,16,160,160)).cuda()   # if modality=='Flow', please change the 2nd dimension 3==>2
    
    input_tensor = torch.from_numpy(img_list)   # transformed numpy array to Tensor
    data=torch.autograd.Variable(input_tensor).cuda()   # if modality=='Flow', please change the 2nd dimension 3==>2
    out=model(data)
    print("type(out):", type(out) )
    print (out.size(),out)
    out_array = out.data.cpu().numpy()  # transformed Variable Tensor to numpy array
    print("The index for max probability: ")
    print(np.argmax(out_array))  # find the biggest probability corresponding the index.

**Comment by qijiezhao** [comment]
You may forget some preprocessing steps:

the image data should first resized to 160*160, then normalize. I didn't find these in your code.

**Comment by xhappy** [comment]
Hi Qijie,

I don't paste the codes here about resizing the picture to 160*160, I had resized the picture in the function "get_img_list_from_vid_reader" as below:

def get_img_list_from_vid_reader(vid, extension, reshape_size=(160, 160), MAX_N_FRAME=16):
    fps = int(round(vid.get_meta_data()['fps']))
    img_list = []

    for i in range(MAX_N_FRAME):
        num = i * fps
        try:
            img = vid.get_data(num)
        except:
            return np.array(img_list)

        if extension == '3gp':
            img = img[25:120, :]  # cropping

        img = cv2.resize(img, reshape_size).astype(np.float32)
        img_list.append(img)

    return np.array(img_list)

**Comment by qijiezhao** [comment]
So normalization step is lost, I suggest you add some functions as bellows:
 [torchvision.transforms.Normalize()](http://pytorch.org/docs/master/torchvision/transforms.html?highlight=normalize)

Another, I don't understand img[25:120] have done. is img a 2d array?
